tree/dataframe/src/RNTupleDS.cxx

```diff
@@ -119,6 +119,55 @@ class RRDFCardinalityField : public ROOT::Experimental::Detail::RFieldBase {
    }
 };
 
+/**
+ * @brief An artificial field that provides the size of a fixed-size array
+ *
+ * This is the implementation of `R_rdf_sizeof_column` in case `column` contains
+ * fixed-size arrays on disk.
+ */
+class RArraySizeField final : public ROOT::Experimental::Detail::RFieldBase {
+private:
+   std::size_t fArrayLength;
```

Comment:
I wonder if that always should be 64bit, like for `ClusterSize_t` in `RRDFCardinalityField`

Reply:
I am not sure, even in `RRDFCardinalityField` most logic uses `std::size_t`. The only exception is in the `GenerateColumnsImpl` method. So I don't think this needs to be changed for `RArraySizeField`?

OTOH, should I also implement `GenerateColumnsImpl` for this class? Currently it does nothing and tests seem to work fine even when using the `sizeof` RDF columns